assets/js/app.js

```diff
@@ -0,0 +1,277 @@
+                        fiveCoinsInStorage.splice(indexToRemove, 1)
+                    }
+                    localStorage.setItem(`5coinsInStorage`, JSON.stringify(fiveCoinsInStorage)) //save
+                })
+            })
+        }
+        allTheFunctionsForCoinPage()
+
+        //the search option
+        const search = document.getElementById("searchButton")
+        search.addEventListener(`click`, async function () {
+            const searchInput = document.getElementById("searchInput").value
+
+            const coinsCache = JSON.parse(localStorage.getItem(`coinsCache`))
+            const index = coinsCache.findIndex(coin => coin.id === searchInput)
+            let data = ``
+            if (index !== -1) data = coinsCache[index]
+            else {
+                const loadingDiv = document.getElementById("loadingDiv")
+                loadingDiv.classList.add(`visible`)
+                data = await getSingleCoin(searchInput)
+                loadingDiv.classList.remove(`visible`)
+            }
+            if (data.id !== undefined) {
+                document.getElementById("coinsTabPresentation").innerHTML =
+                    `
+                    <div class="card" style="width: 18rem;">
+                       <div class="card-body">
+                            <div class="form-check form-switch">
+                                <input class="form-check-input" type="checkbox" role="switch" id="switch${data.id}" style= "position: absolute; right: 10px; font-size: xxx-large;">
+                            </div>
+                            <h5 class="card-title">${data.name}</h5>
+                            <p class="card-text" >${data.id}</p>
+                            <button type="button" class="btn btn-lg btn-primary" data-bs-toggle="popover" id="${data.id}" data-bs-content=""> More Info </button>
+                        </div>
+                    </div>
+            `
+                allTheFunctionsForCoinPage()
+            }
+            else alert(`could not find a coin by id: ${searchInput}`)
+        })
+        document.getElementById("reports-click-tab").addEventListener(`click`, () => {
+            const buttons = document.querySelectorAll("div > button")
+            buttons.forEach(function (button) {
+                bootstrap.Popover.getInstance(button)?.dispose()  // Dispose of the current popover instance, in case there is such
+            })
+        })
+        document.getElementById("about-click-tab").addEventListener(`click`, () => {
+            const buttons = document.querySelectorAll("div > button")
+            buttons.forEach(function (button) {
+                bootstrap.Popover.getInstance(button)?.dispose()  // Dispose of the current popover instance, in case there is such
+            })
+        })
+    })
+
+
+
+    document.getElementById("reports-click-tab").addEventListener(`click`, async () => {
```

Comment:
couldn't get it to work...
<img width="1037" alt="Screen Shot 2025-01-29 at 13 00 57" src="https://github.com/user-attachments/assets/f43be45a-d7b5-42ce-9c29-3faf34d20f79" />